src/gc-heap-snapshot.cpp

```diff
@@ -403,12 +409,13 @@ void _gc_heap_snapshot_record_internal_array_edge(jl_value_t *from, jl_value_t *
                     g_snapshot->names.find_or_create_string_id("<internal>"));
 }
 
-void _gc_heap_snapshot_record_hidden_edge(jl_value_t *from, void* to, size_t bytes) JL_NOTSAFEPOINT
+void _gc_heap_snapshot_record_hidden_edge(jl_value_t *from, void* to, size_t bytes, uint16_t pooled) JL_NOTSAFEPOINT
 {
     size_t name_or_idx = g_snapshot->names.find_or_create_string_id("<native>");
 
     auto from_node_idx = record_node_to_gc_snapshot(from);
-    auto to_node_idx = record_pointer_to_gc_snapshot(to, bytes, "<malloc>");
+    auto alloc_type = pooled ? "<pooled>" : "<malloc>";
```

Comment:
I don't think pooled is the right term. I think we have inline, malloc, foreign. And I think in the case of inline we should increase the size reported here?

Reply:
Pooled was the name of the flag in the jl_array_t struct. Why would it report the wrong size in this case?

Reply:
Ah we actually have 4 different classes.

1. small arrays are inline allocated
2. medium sized array -> pooled
3. large array -> malloced
4. foreign: Who knows.

Now I remember that when were looking together at some snapshots one some arrays contained a pointer to a data object.

Reply:
Foreign arrays are treated exactly like malloced arrays. I found that out while playing with mimalloc, that's why I needed an extra bit in the array because to make it work one can't assume the same allocator for the foreign arrays as the local ones. We currently ignore the inlined arrays in the snapshot. I will see if I can add them however.

Reply:
![image](https://user-images.githubusercontent.com/28694980/200078520-9c2dd31a-3504-4ad4-8d30-8e4395e51dda.png)
I think I found a couple megabytes sneaking around

Reply:
This actually made it a lot more useful because we can now see the size of the inferred code.

Reply:
nice. Is this RTM?

Reply:
IMO, yes.

Reply:
@vchuravy RTM?